--- mercury/boehm_gc/include/private/gcconfig.h	2003-10-28 19:48:43.000000000 +1100
+++ mercury/boehm_gc/include/private/gcconfig.h	2005-08-26 15:31:40.438114698 +1000
@@ -1818,10 +1818,10 @@
 	     extern int etext[];
 #            define DATASTART ((ptr_t)((((word) (etext)) + 0xfff) & ~0xfff))
 #       endif
-#	define PREFETCH(x) \
-	  __asm__ __volatile__ ("	prefetch	%0": : "m"(*(char *)(x)))
-#	define PREFETCH_FOR_WRITE(x) \
-	  __asm__ __volatile__ ("	prefetchw	%0": : "m"(*(char *)(x)))
+#       if defined(__GNUC__) && __GNUC >= 3
+#           define PREFETCH(x) __builtin_prefetch((x), 0, 0)
+#           define PREFETCH_FOR_WRITE(x) __builtin_prefetch((x), 1)
+#       endif
 #   endif
 # endif
 
